HotorNot/app/src/main/java/com/example/hotornot/fragments/RegistrationScreenFragment.kt

```diff
@@ -1,35 +1,175 @@
+            return "Can't be empty!"
+        }
+        return null
+    }
+
+    private fun isValidEmail(): String? {
+        val emailText = binding.txtEmail.text.toString()
+        if (!Patterns.EMAIL_ADDRESS.matcher(emailText).matches()) {
+            return "Invalid Email Address!"
+        }
+        return null
+    }
+
     private fun showInterests() {
-        val spinnerArrayAdapter: ArrayAdapter<String> = ArrayAdapter<String>(
-            requireContext(), android.R.layout.simple_spinner_item,
-            resources.getStringArray(R.array.interests)
+        val arrayAdapter = ArrayAdapter(requireContext(),
+            android.R.layout.simple_spinner_dropdown_item,
+            resources.getStringArray(R.array.interests))
+        binding.spinnerMenu.adapter = arrayAdapter
+        binding.spinnerMenu.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View?,
+                position: Int,
+                id: Long,
+            ) {
+                Toast.makeText(requireContext(),
+                    "You select " + resources.getStringArray(R.array.interests)[position],
+                    Toast.LENGTH_LONG).show()
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+                Toast.makeText(requireContext(),
+                    "Please select one of our options!",
+                    Toast.LENGTH_LONG).show()
+            }
+        }
+    }
+
+    private fun createUser() {
+        val firstName = binding.firstName.text.toString()
+        val lastName = binding.lastName.text.toString()
+        val email = binding.txtEmail.text.toString()
+
+        val user = User(
+            firstName,
+            lastName,
+            email,
+            getSelectRadioBtnValue(), ""
         )
-        spinnerArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
-        spinnerMenu.adapter = spinnerArrayAdapter
+        preferencesUtil.saveUserData(user)
+    }
+
+    private fun getSelectRadioBtnValue() =
+        when (binding.rgGender.checkedRadioButtonId) {
+            R.id.btnRadioMan -> Gender.Male
+            R.id.btnRadioWoman -> Gender.Female
+            R.id.btnRadioOther -> Gender.Other
+            else -> Gender.Other
+        }
+
+    private fun isRegisterUser() {
+        binding.btnReg.setOnClickListener {
+            isItEmpty()
+        }
+    }
+
+    private fun isItEmpty() {
+        val firstName = binding.txtFirstName.editText?.text.toString()
+        val lastName = binding.txtLastName.editText?.text.toString()
+        val email = binding.txtEmailReg.editText?.text.toString()
+
+        if (TextUtils.isEmpty(firstName) || TextUtils.isEmpty(lastName) || TextUtils.isEmpty(
+                email)
+        ) {
+            Toast.makeText(this.context, "Field is required!", Toast.LENGTH_SHORT).show()
```

Comment:
Create own method for toast messages and use it everywhere,Just change the behavior.